packages/grid/_modules_/grid/components/cell/GridEditDateCell.tsx

```diff
@@ -19,7 +19,7 @@ const useUtilityClasses = (ownerState: OwnerState) => {
   return composeClasses(slots, getDataGridUtilityClass, classes);
 };
 
-export function GridEditDateCell(props: GridRenderEditCellParams & InputBaseProps) {
+export function GridEditDateCell(props: GridRenderEditCellParams & Omit<InputBaseProps, 'id'>) {
```

Comment:
I do not understand why removing the `id` props from the signature. Is it because you want to force `id` to get its type from `GridRenderEditCellParams`?

Reply:
Both `GridRenderEditCellParams` and `InputBaseProps` have an `id` prop.

`GridRenderEditCellParams['id'] === GridRowId`
`InputBaseProps['id`] === string | undefined`

So now that we are typing the `api` param, TypeScript is complaining that we are trying to pass `undefined` to the `id` param on `api.setEditCellValue({ id, field, value: newParsedDate }, event);`.

And the way the component is built, the `id` prop will never be passed as `id` to the HTML element (nor should it be).
So I forced the typing to set the `id` prop as the one from `GridRenderEditCellParams` to correctly reflect the current behavior.